faker/providers/address/en_NZ/__init__.py

```diff
@@ -0,0 +1,256 @@
+        'Range Road', 'Range Road',
+        'Road', 'Road', 'Road', 'Road', 'Road', 'Road', 'Road', 'Road', 'Road',
+        'Street', 'Street', 'Street', 'Street', 'Street', 'Street', 'Street',
+        'Street', 'Street', 'Street', 'Street', 'Street', 'Street', 'Street',
+        'Street', 'Street', 'Street', 'Street', 'Street', 'Street', 'Street',
+        'Terrace', 'Terrace', 'Terrace',
+        'Way', 'Way', 'Way',
+
+        # Other:
+        'Access', 'Alley', 'Alleyway', 'Amble', 'Anchorage', 'Approach',
+        'Broadway', 'Bypass', 'Causeway', 'Centre',
+        'Circle', 'Circuit', 'Close', 'Concourse', 'Copse', 'Corner', 'Court',
+        'Cove',
+        'Crest', 'Cross', 'Crossing',
+        'Cutting',
+        'Esplanade',
+        'Flats',
+        'Gardens', 'Grove', 'Heights', 'Highway',
+        'Lane', 'Line', 'Keys',
+        'Parade', 'Park', 'Pass',
+        'Plaza',
+        'Point', 'Quay',
+        'Reserve',
+        'Ridge',
+        'Rise',
+        'Square',
+        'Track', 'Trail',
+        'View',
+    )
+
+    te_reo_parts = (
+        'ara',
+        'awa',
+        'horo',
+        'kawa',
+        'koro',
+        'kowhai',
+        'manawa',
+        'mata',
+        'maunga',
+        'moko',
+        'motu',
+        'ngauru',
+        'pa'
+        'papa',
+        'po',
+        'puke',
+        'rangi',
+        'rohe',
+        'rongo',
+        'roto',
+        'tahi',
+        'tai',
+        'tangi',
+        'tau',
+        'tere',
+        'tipu',
+        'wai',
+        'waka',
+        'whaka',
+        'whanga',
+        'whare',
+        'weka',
+    )
+
+    te_reo_endings = (
+        'hanga',
+        'hope',
+        'iti',
+        'iti',
+        'kiwi',
+        'makau',
+        'nui',
+        'nui',
+        'nui',
+        'nuku',
+        'roa',
+        'rua',
+        'tanga',
+        'tapu',
+        'toa',
+        'whenua',
+        'whero',
+        'whitu',
+    )
+
+    postcode_formats = (
+        # as per https://en.wikipedia.org/wiki/Postcodes_in_New_Zealand
+        # Northland
+        '01##',
+        '02##',
+        '03##',
+        '04##',
+        '05##',
+        '06##',
+        '07##',
+        '08##',
+        '09##',
+        # Auckland
+        '1###',
+        '20##',
+        '21##',
+        '22##',
+        '23##',
+        '24##',
+        '25##',
+        '26##',
+        # Central North Island
+        '3###',
+        '4##',
+        # Lower North Island
+        '50##',
+        '51##',
+        '52##',
+        '53##',
+        '55##',
+        '57##',
+        '58##',
+        # Wellington
+        '60##',
+        '61##',
+        '62##',
+        '64##',
+        '69##',
+        # Upper South Island
+        '7###',
+        # Christchurch
+        '80##',
+        '81##',
+        '82##',
+        '84##',
+        '85##',
+        '86##',
+        '88##',
+        '89##',
+        # The bottom of the Wurrrld
```

Comment:
No LOL please :) Does this administrative area have a name? If I'm reading wikipedia right, it should be Southland

Reply:
ah - whoops :-P